app/models/product.rb

```diff
@@ -8,15 +8,34 @@ class Product < ApplicationRecord
   validates :title, :description, :image_url, presence: true
 # 
   validates :title, uniqueness: true
-  validates :image_url, allow_blank: true, format: {
-    with:    %r{\.(gif|jpg|png)\z}i,
-    message: 'must be a URL for GIF, JPG or PNG image.'
-  }
+
+  # Validate image_url format by using url:true
+  validates :image_url, allow_blank: true, url: true 
+
   validates :title, length: {minimum: 10}
-  validates :price, numericality: { greater_than_or_equal_to: 0.01 }
 
+  #Price numericality validation should only run if price is present.
+  validates :price, numericality: { greater_than_or_equal_to: 0.01 }, if: :price?
+
+  # Permalink must be unique and no special character and no space allowed and minimum 3 words, separated by hyphen
+  validates :permalink, uniqueness: true, format: { with: PERMALINK_REGEX}
+
+  # Description - should be between 5 and 10 words
+  validates :description, format: { with: DESCRIPTION_REGEX}
+
+  #using custom validator
+  validate :price_should_be_greater_than_discount_price
+
+  #without custom validator
+  validates :price, numericality: { greater_than: :discount_price }, if: :discount_price?
```

Comment:
what if `discount_price` is negative, ex: -2, then price can be -1, as that valid?